packages/compass-preferences-model/lib/model.js

```diff
@@ -1,323 +1,273 @@
-    debug('app page refresh detected');
-  },
-  onAppRestart: function() {
-    debug('app restart detected');
+class Preferences {
+  constructor(userDataPath) {
+    // User preferences are stored to disc via the Ampersand model.
+    const PreferencesModel = Model.extend(storageMixin, {
+      props: preferencesProps,
+      extraProperties: 'ignore',
+      idAttribute: 'id',
+      namespace: 'AppPreferences',
+      storage: {
+        backend: 'disk',
+        basepath: userDataPath // userDataPath of the electron app.
+      }
+    });
+
+    this.userPreferencesModel = new PreferencesModel();
+  }
+
+  fetchPreferences() {
+    return new Promise((resolve, reject) => {
+      // Fetch user preferences from the Ampersand model.
+      this.userPreferencesModel.fetch({
+        success: (model) => {
+          debug('fetch user preferences is successful', model.serialize());
+          return resolve(this.getAllPreferences());
+        },
+        error: (model, err) => {
+          debug('fetch user preferences failed', err.message);
+          return reject(err);
+        }
+      });
+    });
+  }
+
+  savePreferences(attributes) {
+    return new Promise((resolve, reject) => {
+      if (!attributes && isEmpty(attributes)) {
+        return resolve(this.getAllPreferences());
+      }
+
+      // Save user preferences to the Ampersand model.
+      this.userPreferencesModel.save(attributes, {
+        success: () => {
+          return resolve(this.getAllPreferences());
+        },
+        error: (model, err) => {
+          debug('saving user preferences error', err);
+          return reject(err);
+        }
+      });
+    });
+  }
+
+  getAllPreferences() {
+    // TODO: merge user, global, and CLI preferences here.
+    return this.userPreferencesModel.getAttributes({ props: true, derived: true });
+  }
+
+  async getConfigurableUserPreferences() {
+    // Set the defaults and also update showedNetworkOptIn flag.
+    if (!this.getAllPreferences().showedNetworkOptIn) {
+      await this.savePreferences({
+        autoUpdates: true,
+        enableMaps: true,
+        trackErrors: true,
+        trackUsageStatistics: true,
+        enableFeedbackPanel: true,
+        showedNetworkOptIn: true,
+        theme: THEMES.LIGHT,
+      });
+    }
+    const preferences = this.getAllPreferences();
+    return Object.fromEntries(
+      Object.entries(preferences).filter(([key]) => preferencesProps[key].ui === true)
+    );
+  }
+}
+
+/**
+ * API to communicate with preferences from the electron renderer process.
+ */
+const preferencesIpc = {
+  savePreferences(attributes) {
+    if (ipc && ipc.ipcRenderer && typeof ipc.ipcRenderer.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:save-preferences', attributes);
+    }
   },
-  onAppVersionMismatch: function(lastKnownVersion, currentVersion) {
-    debug('version mismatch detected: was %s, now %s',
-      lastKnownVersion, currentVersion);
+  getPreferences() {
```

Comment:
Is the naming mismatch with `Preferences` intentional (getPreferences vs getAllPreferences)?